### PR TITLE
feat: support system proxy

### LINKED ISSUE #407: feature request: system proxy support
My PC have to use proxy to access GitHub (you know why)

```bash
git node --proxy $URL land $PR
# or
git node set proxy="$URL" # set proxy to config
# or
git node --proxy land $PR # use proxy in config
```

and I will take some time to handle this

### PR DESCRIPTION
**Pull request by Himself65**
FIxes #407 

- [x] Support set proxy from `git config --get https.proxy`
- [x] Support set proxy from `ncu-config`

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -6,8 +6,9 @@ const path = require('path');
 const { CI_DOMAIN } = require('./ci/ci_type_parser');
 
 class Request {
-  constructor(credentials) {
+  constructor({ credentials, proxyAgent = undefined } = {}) {
```
Comment (joyeecheung) [inline lib/request.js:9]: Why not just create a proxyAgent here (at the top level so it's shared) and use it throughout the code base?

File: lib/proxy.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const { spawnSync } = require('child_process');
+const {
+  getConfig, GLOBAL_CONFIG,
+  PROJECT_CONFIG, LOCAL_CONFIG
+} = require('./config');
+
+function proxy(proxyUrl) {
+  if (proxyUrl == null || proxyUrl === '') {
+    proxyUrl =
+      getConfig(GLOBAL_CONFIG).proxy ||
```
Comment (joyeecheung) [inline lib/proxy.js:12]: You can just use `require('../../lib/config').getMergedConfig().proxy` here

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "figures": "^3.1.0",
     "fs-extra": "^8.1.0",
     "ghauth": "^4.0.0",
+    "https-proxy-agent": "^5.0.0",
```
Comment (joyeecheung) [inline package.json:44]: Why not use https://www.npmjs.com/package/proxy-agent to support more protocols?

**Comment by codebytere** [comment]
@Himself65 can you rebase this?

**Comment by Himself65** [comment]
ok. i will do this later

**Pull request merged**